sessionctx/variable/session.go

```diff
@@ -1575,9 +1575,31 @@ func (s *SessionVars) GetSplitRegionTimeout() time.Duration {
 	return time.Duration(s.WaitSplitRegionTimeout) * time.Second
 }
 
+// ContainSpecialIsolationRead checks whether we can access the special read engines.
+func (s *SessionVars) ContainSpecialIsolationRead(engineType kv.StoreType) bool {
+	_, ok := s.isolationReadEngines[engineType]
+	return ok
+}
+
 // GetIsolationReadEngines gets isolation read engines.
+// The return value is read-only.
```

Comment:
It seems that we create a copy of `s.isolationReadEngines` and modifying the copy doesn't affect `s.isolationReadEngines`. Why do we need to address that it is read-only?

Reply:
/cc @qw4990

Reply:
Maybe the comment is misleading. Just want to say the change in the outside will not affect the original value. I'll fix it.